doc/setup.md

```diff
@@ -0,0 +1,286 @@
+
+```
+tapyrus-setup createkey
+
+output <ephemeral_private_key[i]> <ephemeral_public_key[i]>
+```
+
+## Step 7. Generate block vss.
+
+As in Step 3, Signer[i] creates "ephemeral" VSS(verifiable secret share).
+
+```
+
+tapyrus-setup createvss -public_key=1:<ephemeral_public_key[1]> -public_key=2:<ephemeral_public_key[2]> ... -public_key=n:<ephemeral_public_key[n]> -private_key=<ephemeral_private_key[i]>
+
+output: {
+    <ephemeral_public_key[1]>: <ephemeral_vss[i, 1]>,
+    <ephemeral_public_key[2]>: <ephemeral_vss[i, 2]>,
+    ...
+    <ephemeral_public_key[n]>: <ephemeral_vss[i, n]>,
+}
+
+```
+
+Signer[i] does not have to specify a nonce used in the encryption. Nonce is optional in this step.
+
+`ephemeral_vss[i, j]` contains the following information:
+
+- the public key `ephemeral_public_key[i]` which indicates the signer who sends this vss.
+- the public key `ephemeral_public_key[j]` which indicates the signer to be received the `ephemeral_vss[i, j]`
+- "positive"[^6] public commitments `commitments[k]` (k = 1, 2, ..., t)
+- "positive" secret `secret[j]` to perform secret sharing scheme.
+- "negative" public commitments `commitments[k]` (k = 1, 2, ..., t)
+- "negative" secret `secret[j]` to perform secret sharing scheme.
+
+`ephemeral_public_key[i, j]` is encrypted in the same way as `static_vss` described in Step 3.
+
+Signer[i] sends the generated values `ephemeral_vss[i, j]` to other signers.
+
+[^6]: In schnorr signature schema used in Tapyrus Core, we use the random value "r" and the ephemeral point "R" on the elliptic curve, where R = rG (G is the generator of the curve). In Tapyrus signature schema we should choose R so that jacobi(y(R)) = 1 (if not, use (n - r) instead of r and generate R so that R = (n - r)G, where n is the order of the curve). So in this step, we should generate both "positive" secret and commitments that corresponds to r, and "negative" ones that corrensponds to (n - r). For more information, see [Tapyrus Schnorr Signature Specification](https://github.com/chaintope/tapyrus-core/blob/master/doc/tapyrus/schnorr_signature.md)
+
+## Step 8. Sign the genesis block locally.
+
+Signer[i] secretly receives `ephemeral_vss[j, i]` generated by other signers. Then Signer[i] generates `local_sig` using `tapyrus-setup sign`.
+To generate `local_sig`, Signer[i] need at least t valid `ephemeral_vss[j, i]`s. In the following steps, we suppose that Signer[1], Signer[2], ... Signer[t] are honest signers providing a valid vss and Signer[i] got `ephemeral_vss[j, i]`(j = 1, 2, 3, ..., t).
+
+```
+tapyrus-setup sign -static_vss=1:<static_vss[1, i]> -static_vss=2:<static_vss[2, i]> ... -static_vss=t:<static_vss[t, i]> -ephemeral_vss=1:<ephemeral_vss[1, i]> -ephemeral_vss=2:<ephemeral_vss[2, i]> ... -ephemeral_vss=t:<ephemeral_vss[t, i]>   -aggregated_public_key=<aggregated_public_key> -nodesecret=<node_secret[i]> -private_key=<static_private_key[i]> -block=<block>
+
+output: <local_sig[i]>
+```
```

Comment:
Actually I don't understand what is trouble.
BTW, I suspect that creating LocalSig doesn't need `static_vss` because aggregated public key and shared secret `node_secret` generated in Key Generation Protocol are supplied as the command arguments.

Reply:
Oh, sorry i missed. What i wanted to say is `ephemeral_vss`.

And I agree with the local sig generation code doesn't need `static_vss`.

Reply:
6cdf1f836de70ce26681e1d606a0e7c5e564c60c